lock-api/src/main/java/com/palantir/lock/client/LockRefreshingLockService.java

```diff
@@ -183,4 +200,23 @@ public void close() {
     public Set<LockRefreshToken> getAllRefreshingTokens() {
         return ImmutableSet.copyOf(toRefresh);
     }
+
+    @FunctionalInterface
+    public interface FailedRefreshCallback {
+        void onFailedRefresh(Set<LockRefreshToken> tokens);
+    }
+
+    /**
+     * Register a callback function to get called any time tokens fail to be refreshed.
+     */
+    public void registerRefreshFailedCallback(FailedRefreshCallback callback) {
+        failedRefreshCallbacks.add(callback);
+    }
+
+    /**
+     * Remove a callback function that was previously registered to be called when tokens fail to be refreshed.
+     */
+    public void removeRefreshFailedCallback(FailedRefreshCallback callback) {
```

Comment:
Might be cleaner to return a `Disposable` from the registration function to unregister callbacks, imo the code is easier to follow and debug that way as opposed to reference-equality register/unregister functions.

Reply:
Done